bundles/alpha.model/src/alpha/model/transformation/automation/OptimalSimplifyingReductions.xtend

```diff
@@ -489,6 +512,14 @@ class OptimalSimplifyingReductions {
 	// Miscellaneous helper functions
 	////////////////////////////////////////////////////////////
 	
+	private def DynamicProgrammingStep getDPStepForSpecialSR(AbstractReduceExpression targetRE, long[] vector) {
+		val space = targetRE.body.contextDomain.copy.toIdentityMap.space
+		val nbParams = space.dim(ISLDimType.isl_dim_param)
+		val zeros = (0..<nbParams).map[0L]
+		val specialRho = space.createUniformFunction(zeros + vector)
+		return new StepRemoveEmbedding(targetRE, specialRho)
+	}
+	
```

Comment:
New dynamic programming step to actually rewrite the reduction and exploit the identical answers.